File: Data Structures & Algorithms/valid-sudoku/submission-1.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& b) {
        
        
        bool r[9][9] = {false}, c[9][9] = {false}, bx[9][9] = {false};

        for(int i=0; i<9; i++){
            for(int j=0; j<9; j++){

                if( b[i][j]=='.' ) continue;

                int v = b[i][j]-'1';

                int bn = (i/3)*3+(j/3);

                if( r[i][v] || c[j][v] || bx[bn][v] ) return false;

                r[i][v] = c[j][v] =  bx[bn][v] = true;

            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/non-cyclical-number/submission-1.java
class Solution {
    public boolean isHappy(int n) {
        
        Set<Integer> s = new HashSet<>();

        while(n>1){

            int sum = 0;

            while(n>0){
                int d = n%10;
                sum+= (d*d);
                n/=10;
            }
            n = sum;
            if( s.contains(sum) ) return false;
            if(sum==1) return true;
            s.add(sum);
        }
        return n==1;
    }
}
